### PR TITLE
feat: Add test for meshsync

### PR DESCRIPTION
**Pull request by Aisuko**
**Description**

This PR is related to #463 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

```
@Aisuko ➜ /workspaces/meshery-operator (feat/meshsync) $ make coverage
make bin/setup-envtest
make[1]: Entering directory '/workspaces/meshery-operator'
go: downloading sigs.k8s.io/controller-runtime v0.13.1-0.20221212190805-d4f1e822ca11
go: downloading sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20221212190805-d4f1e822ca11
go: downloading github.com/go-logr/zapr v1.2.0
go: downloading github.com/go-logr/logr v1.2.0
go: downloading github.com/spf13/afero v1.6.0
go: downloading go.uber.org/zap v1.19.1
go: downloading golang.org/x/text v0.3.7
go: downloading go.uber.org/atomic v1.7.0
make[1]: Leaving directory '/workspaces/meshery-operator'
bin/setup-envtest use 1.24.2 --bin-dir /workspaces/meshery-operator/bin
Version: 1.24.2
OS/Arch: linux/amd64
md5: ZViX8vb02m8ULK0fkiNNGw==
Path: /workspaces/meshery-operator/bin/k8s/1.24.2-linux-amd64
go test -v ./... -coverprofile cover.out
?       github.com/layer5io/meshery-operator    [no test files]
=== RUN   TestAPIs
Running Suite: APIs Suite - /workspaces/meshery-operator/api/v1alpha1
=====================================================================
Random Seed: 1684734948

Will run 12 of 12 specs
••••••••••••

Ran 12 of 12 Specs in 0.005 seconds
SUCCESS! -- 12 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAPIs (0.01s)
PASS
        github.com/layer5io/meshery-operator/api/v1alpha1       coverage: 49.2% of statements
ok      github.com/layer5io/meshery-operator/api/v1alpha1       0.020s  coverage: 49.2% of statements
?       github.com/layer5io/meshery-operator/pkg/client [no test files]
?       github.com/layer5io/meshery-operator/pkg/client/v1alpha1        [no test files]
=== RUN   TestAPIs
Running Suite: Controller Suite - /workspaces/meshery-operator/controllers
==========================================================================
Random Seed: 1684734951

Will run 11 of 11 specs
Ran 11 of 11 Specs in 66.117 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAPIs (66.12s)
PASS
        github.com/layer5io/meshery-operator/controllers        coverage: 48.8% of statements
ok      github.com/layer5io/meshery-operator/controllers        66.152s coverage: 48.8% of statements
=== RUN   TestAPIs
Running Suite: Broker Suite - /workspaces/meshery-operator/pkg/broker
=====================================================================
Random Seed: 1684734951

Will run 3 of 3 specs
•••

Ran 3 of 3 Specs in 5.486 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAPIs (5.49s)
PASS
        github.com/layer5io/meshery-operator/pkg/broker coverage: 73.7% of statements
ok      github.com/layer5io/meshery-operator/pkg/broker 5.519s  coverage: 73.7% of statements
=== RUN   TestGetObjects
--- PASS: TestGetObjects (0.00s)
PASS
        github.com/layer5io/meshery-operator/pkg/meshsync       coverage: 100.0% of statements
ok      github.com/layer5io/meshery-operator/pkg/meshsync       0.007s  coverage: 100.0% of statements
go tool cover -html=cover.out -o cover.html
```

### CODE REVIEW

**Review by Micost — APPROVED** [review]

**Review by Micost — APPROVED** [review]

**Review by Micost — COMMENTED** [review]
approved and ready to go

**Review by Azanul — CHANGES_REQUESTED** [review]

File: Makefile [hunk 1]
```diff
@@ -271,6 +271,7 @@ catalog-push: ## Push a catalog image.
 
 # Test coverage
 .PHONY: coverage
+coverage: test-env
 coverage: ## Generate test coverage report
```
[the remark this answers is not in the document]
Comment (Aisuko) [reply Makefile:275]: The first one is used to `test-env`, and the second one is used to run the testing.

[the remark this answers is not in the document]
Comment (Azanul) [reply Makefile:275]: Why not call test-env under the already existing target; suggesting this only for readability as duplicate targets can sometimes create confusion.

[the remark this answers is not in the document]
Comment (Aisuko) [reply Makefile:275]: For sure, why not? Do you mean like this?
```
coverage:
    test-env
    go test -v 

```

[the remark this answers is not in the document]
Comment (Azanul) [reply Makefile:275]: Yes

[the remark this answers is not in the document]
Comment (Azanul) [reply Makefile:275]: Do not miss the `go tool` line there

[the remark this answers is not in the document]
Comment (Aisuko) [reply Makefile:275]: ok, thank you. It is reasonable. Will do soon.

**Review by Azanul — APPROVED** [review]

**Pull request merged**